ocaml/xapi/message_forwarding.ml

```diff
@@ -629,6 +629,14 @@ functor
           Ref.string_of repository
       with _ -> "invalid"
 
+    let observer_uuid ~__context observer =
+      try
+        if Pool_role.is_master () then
+          Db.Observer.get_uuid ~__context ~self:observer
+        else
+          Ref.string_of observer
+      with _ -> "invalid"
```

Comment:
Wouldn't it be better if it failed here instead of later on?

In any case it looks like this only ever gets used in the coordinator, so the check is not useful. At which point it seems like it might be worth to just use get_uuid in the functions themselves.

If there's a good reason to avoid failure at this point ignore this

Reply:
Looking above, this is the standard for `*_uuid` with the DB object.